ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -1,458 +1,470 @@
 {%             elif router_bgp.address_family_ipv4_multicast.neighbors[neighbor].activate == false %}
       no neighbor {{ neighbor }} activate
 {%             endif %}
-{%          endfor %}
-{%       endif %}
-{%       if router_bgp.address_family_ipv4_multicast.redistribute_routes is defined and router_bgp.address_family_ipv4_multicast.redistribute_routes is not none %}
+{%         endfor %}
 {%         for redistribute_route in router_bgp.address_family_ipv4_multicast.redistribute_routes | arista.avd.natural_sort %}
-      redistribute {{ redistribute_route }} {% if router_bgp.address_family_ipv4_multicast.redistribute_routes[redistribute_route].route_map is defined and router_bgp.address_family_ipv4_multicast.redistribute_routes[redistribute_route].route_map is not none %}route-map {{ router_bgp.address_family_ipv4_multicast.redistribute_routes[redistribute_route].route_map }}{% endif %}
-
+{%             set redistribute_route_cli = "redistribute " ~ redistribute_route %}
+{%             if router_bgp.address_family_ipv4_multicast.redistribute_routes[redistribute_route].route_map is arista.avd.defined %}
+{%                 set redistribute_route_cli = redistribute_route_cli ~ " route-map " ~ router_bgp.address_family_ipv4_multicast.redistribute_routes[redistribute_route].route_map %}
+{%             endif %}
+      {{ redistribute_route_cli }}
 {%         endfor %}
-{%       endif %}
 {%     endif %}
 {# address family ipv6 activation #}
-{%     if router_bgp.address_family_ipv6 is defined and router_bgp.address_family_iv6 is not none %}
+{%     if router_bgp.address_family_ipv6 is arista.avd.defined %}
    !
    address-family ipv6
-{%       if router_bgp.address_family_ipv6.peer_groups is defined and router_bgp.address_family_ipv6.peer_groups is not none %}
-{%          for peer_group in router_bgp.address_family_ipv6.peer_groups | arista.avd.natural_sort %}
-{%             if router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_in is defined and router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_in is not none %}
+{%         for peer_group in router_bgp.address_family_ipv6.peer_groups | arista.avd.natural_sort %}
+{%             if router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_in is arista.avd.defined %}
       neighbor {{ peer_group }} route-map {{ router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_in }} in
 {%             endif %}
-{%             if router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_out is defined and router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_out is not none %}
+{%             if router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_out is arista.avd.defined %}
       neighbor {{ peer_group }} route-map {{ router_bgp.address_family_ipv6.peer_groups[peer_group].route_map_out }} out
 {%             endif %}
 {%             if router_bgp.address_family_ipv6.peer_groups[peer_group].activate == true %}
```

Comment:
No check if `router_bgp.address_family_ipv6.peer_groups[peer_group].activate` is defined